books-call-number/nodes/js/buildQuery.js

```diff
@@ -16,22 +28,26 @@ if (endRange) {
 
 where += '\n\t\tAND ie.status_name = \'Checked out\'';
 
+if (locationNameArray.length > 0 && locationNameArray.length < totalCharacterCount) {
```

Comment:
locationNameArray.length < totalCharacterCount is always going to be less than the total number of characters summing the location names.
```suggestion
if (locationNameArray.length > 0 && totalCharacterCount  <= maxAllowedCharacters ) {
```